avm/src/daemon/ctrl.rs

```diff
@@ -424,6 +425,33 @@ async fn dsgetv_inner(req: Request<Body>) -> DaemonResult<Arc<HandlerMemory>> {
   Ok(hand_mem)
 }
 
+async fn handle_dsgetr(req: Request<Body>) -> Result<Response<Body>, Infallible> {
+  match dsgetr_inner(req).await {
+    Ok(hand_mem) => {
+      let mut out = vec![];
+      hand_mem.to_pb().write_to_vec(&mut out).unwrap();
+      Ok(Response::builder().status(200).body(out.into()).unwrap())
+    }
+    Err(err) => {
+      // TODO: What error message here? Also should this also be a valid HM out of here?
+      eprintln!("{:?}", err);
+      Ok(Response::builder().status(500).body("fail".into()).unwrap())
+    }
+  }
+}
+
+async fn dsgetr_inner(req: Request<Body>) -> DaemonResult<Arc<HandlerMemory>> {
+  let bytes = body::to_bytes(req.into_body()).await?;
+  let body: DSGet = serde_json::from_slice(&bytes).unwrap();
```

Comment:
i think it makes sense to not `unwrap` the deserialization and instead bubble up the error with `?` - a malformed JSON structure is a valid error that shouldn't cause a panic

Reply:
That was copy-pasted, so I'll fix that in the others. You're right on this.